mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2426,13 +2426,20 @@ where
     };
     // record secret hash to blockchain too making it impossible to lose
     // lock time may be easily brute forced so it is not mandatory to record it
-    let secret_hash_op_return_script = Builder::default()
-        .push_opcode(Opcode::OP_RETURN)
-        .push_bytes(secret_hash)
-        .into_bytes();
-    let secret_hash_op_return_out = TransactionOutput {
+    let mut op_return_builder = Builder::default().push_opcode(Opcode::OP_RETURN);
+
+    // add the full redeem script to the OP_RETURN for ARRR to simplify the validation for the daemon
+    op_return_builder = if coin.as_ref().conf.ticker == "ARRR" {
```

Comment:
My only question is shouldn't we try to avoid hardcoded tickers? I know this will probably be used for "ARRR" only but I don't know if it's better to add this to the coins config file as maybe "full_redeem_script" or not.

Reply:
I will change it in the next iterations, there will be a custom version of `swap_payment_outputs` for ZCoin or maybe another abstraction.